src/platform/forms-system/src/js/helpers.js

```diff
@@ -173,18 +177,18 @@ export function filterViewFields(data) {
     if (Array.isArray(field)) {
       const newArray = field.map(item => filterViewFields(item));
 
-      return _.set(nextProp, newArray, newData);
+      return set(nextProp, newArray, newData);
     }
 
     if (typeof field === 'object') {
       if (nextProp.startsWith('view:')) {
-        return _.assign(newData, filterViewFields(field));
+        return { ...newData, ...filterViewFields(field) };
       }
-      return _.set(nextProp, filterViewFields(field), newData);
+      return set(nextProp, filterViewFields(field), newData);
```

Comment:
`newData` is the reducer accumulator, so mutating it the old fashioned way with an assignment and then returning would be good.

```js
newData[nextProp] = filterViewFields(field);
return newData;
```